test/jdk/com/sun/net/httpserver/bugs/B6431193.java

```diff
@@ -40,59 +40,48 @@
 
 public class B6431193 {
 
-    static boolean error = false;
+    static boolean handlerIsDaemon = false;
 
-    public static void read (InputStream i) throws IOException {
-        while (i.read() != -1);
-        i.close();
-    }
-
-    /**
-     * @param args
-     */
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         class MyHandler implements HttpHandler {
             public void handle(HttpExchange t) throws IOException {
-                InputStream is = t.getRequestBody();
-                read(is);
-                // .. read the request body
+                try (InputStream is = t.getRequestBody();
+                     OutputStream os = t.getResponseBody()) {
+                    is.readAllBytes();
+                    // .. read the request body
                     String response = "This is the response";
-                t.sendResponseHeaders(200, response.length());
-                OutputStream os = t.getResponseBody();
-                os.write(response.getBytes());
-                os.close();
-                error = Thread.currentThread().isDaemon();
+                    t.sendResponseHeaders(200, response.length());
+                    os.write(response.getBytes());
+                    os.close();
+                    handlerIsDaemon = Thread.currentThread().isDaemon();
+                }
             }
         }
 
-
-        HttpServer server;
+        InetAddress loopback = InetAddress.getLoopbackAddress();
+        HttpServer server = HttpServer.create(new InetSocketAddress(loopback, 0), 10);
         try {
-            InetAddress loopback = InetAddress.getLoopbackAddress();
-            server = HttpServer.create(new InetSocketAddress(loopback, 0), 10);
-
             server.createContext("/apps", new MyHandler());
             server.setExecutor(null);
             // creates a default executor
-                server.start();
+            server.start();
```

Comment:
server.start() (and what preceeds) should be called before entering the try block